src/Mimi.jl

```diff
@@ -571,6 +599,50 @@ function run_timestep(s::adder, t::Int)
     v.output[t] = p.input[t] + p.add[t]
 end
 
+#Begin Graph Functionality section
+
+function show(io::IO, m::Model)
+    println(io, "showing model component connections:")
+    i=1
+    for c in keys(m.components2)
```

Comment:
You can use the [``enumerate``](http://docs.julialang.org/en/release-0.4/stdlib/collections/?highlight=enumerate#Base.enumerate)  function to give you the ``c`` and the ``i`` here.